helix-view/src/file_watcher.rs

```diff
@@ -0,0 +1,97 @@
+use crossbeam_channel::unbounded;
+use crossbeam_channel::{select, Receiver};
+use notify::{recommended_watcher, RecommendedWatcher, RecursiveMode, Watcher};
+use std::path::PathBuf;
+use tokio::sync::mpsc::{unbounded_channel, UnboundedReceiver};
+
+pub struct NotifyActor {
+    sender: Sender,
+    watcher: (RecommendedWatcher, Receiver<NotifyEvent>),
+}
+
+impl NotifyActor {
+    pub fn spawn() -> NotifyHandle {
+        let (actor_tx, actor_rx) = crossbeam_channel::unbounded();
+        let (tx, rx) = unbounded_channel();
+        let actor = NotifyActor::new(Box::new(move |msg| tx.send(msg).unwrap()));
+        let thread = jod_thread::Builder::new()
```

Comment:
Let's drop the jod_thread dependency and instead just call `.join` at the appropriate spot. You probably want to close the channels then join after the run loop terminates https://github.com/helix-editor/helix/blob/05f3fa69e9ce402cf27561aba899b5e28c223de9/helix-term/src/application.rs#L542

Reply:
since I switched to tokio tasks they should join when dropping